generator/src/main/java/com/scottlogic/deg/generator/generation/fieldvaluesources/datetime/RandomDateIterator.java

```diff
@@ -25,11 +27,26 @@ public boolean hasNext() {
 
     @Override
     public OffsetDateTime next() {
-        long min = this.minDate.toInstant().toEpochMilli();
-        long max = this.maxDate.toInstant().toEpochMilli() - 1;
+        long min = getMilli(minDate);
+        long max = getMilli(maxDate) - 1;
 
-        long generatedLong = (long)random.nextDouble(min, max);
+        long generatedLong = (long) random.nextDouble(min, max);
 
-        return Instant.ofEpochMilli(generatedLong).atZone(ZoneOffset.UTC).toOffsetDateTime();
+        OffsetDateTime generatedDate = Instant.ofEpochMilli(generatedLong).atZone(ZoneOffset.UTC).toOffsetDateTime();
+
+        return trimUnwantedGranularity(generatedDate, granularity);
+    }
+
+    private long getMilli(OffsetDateTime date) {
+        return date.toInstant().toEpochMilli();
+    }
+
+    private int nanoToMilli(int nano) {
+        int factor = 1_000;
```

Comment:
`1_000_000` for nano to milli?

Reply:
I agree with this, and have resolved the issue.